File: store-service/internal/pricing/service.go
package pricing

import "duckstore/store-service/internal/service"

// PricingService is the service-shaped adapter over Calculate. The rules
// pipeline stays in the package-level Calculate — the service just exposes
// it as a method for consumers that want to accept a Pricer interface.
type PricingService struct {
	service.BaseService
}

func NewService() *PricingService {
	return &PricingService{BaseService: service.New("pricing")}
}

func (s *PricingService) Calculate(req Request) Result {
	return Calculate(req)
}

File: backend/src/pricing/service.js
import { BaseService } from "../services/BaseService.js";
import { calculate } from "./pricing.js";

// PricingService wraps the Chain-of-Responsibility calculator as a
// service. The rules pipeline stays in `pricing.js`; the service just
// exposes it as a method for consumers that want to inject it.
export class PricingService extends BaseService {
  constructor() {
    super({ entityName: "pricing" });
  }

  calculate(req) {
    return calculate(req);
  }
}
